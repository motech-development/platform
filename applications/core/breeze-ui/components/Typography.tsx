import { createElement, FC, HTMLAttributes } from 'react';

type TAlign = 'left' | 'right' | 'center';

type TComponent = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'p';

type TMargin = 'none' | 'sm' | 'md' | 'lg';

type TVariant = TComponent | 'lead';

const styles = (align: TAlign, margin: TMargin, variant: TVariant) => {
  let alignStyles: string;
  let marginStyles: string;
  let variantStyles: string;

  switch (align) {
    case 'center':
      alignStyles = 'text-center';
      break;
    case 'right':
      alignStyles = 'text-right';
      break;
    default:
      alignStyles = 'text-left';
  }

  switch (margin) {
    case 'sm':
      break;
    case 'md':
      break;
    case 'lg':
      marginStyles = 'mb-6';
      break;
    default:
  }

  switch (variant) {
    case 'h1':
      variantStyles = 'font-display text-5xl font-semibold';
      break;
    case 'h2':
      variantStyles = 'font-display text-3xl font-semibold';
      break;
    case 'h3':
      variantStyles = 'font-display font-semibold';
      break;
    case 'h4':
      variantStyles = 'font-display font-semibold';
      break;
    case 'h5':
      variantStyles = 'font-display font-semibold';
      break;
    case 'h6':
      variantStyles = 'font-display font-semibold';
      break;
    case 'lead':
      variantStyles = 'font-light text-xl';
      break;
    default:
      variantStyles = 'font-sans';
  }

  return `${alignStyles} ${marginStyles} ${variantStyles}`;
};

interface ITypographyProps extends HTMLAttributes<HTMLElement> {
  align?: 'left' | 'right' | 'center';
  component: TComponent;
  margin?: TMargin;
  variant: TVariant;
}

const Typography: FC<ITypographyProps> = ({
  align = 'left',
  children,
  component,
  margin = 'md',
  variant,
  ...rest
}) =>
  createElement(
    component,
    {
      ...rest,
      className: `${styles(align, margin, variant)}`,
    },
    children,
  );

export default Typography;
