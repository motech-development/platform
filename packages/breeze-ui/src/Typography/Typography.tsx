import { FC, memo, ReactNode } from 'react';
import styled from 'styled-components';

type TypographyVariants = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'p';

interface IBaseTypographyProps {
  $align: 'left' | 'right' | 'center';
  $breakWord: boolean;
  $margin: keyof typeof margins;
  $truncate: boolean;
  $variant: TypographyVariants | 'lead';
}

const BaseTypography = styled.p<IBaseTypographyProps>`
  ${({ $align, $breakWord, $margin, $truncate, $variant }) => `
    margin: ${margins[$margin] ? `0 0 ${margins[$margin]}rem` : '0'};
    text-align: ${$align};

    ${
      $breakWord
        ? `
      hyphens: auto;
      overflow-wrap: break-word;
      word-wrap: break-word;
      word-break: break-all;
      word-break: break-word;
    `
        : ''
    }

    ${
      $truncate
        ? `
      overflow: hidden;
      text-overflow: ellipsis;
      white-space: nowrap;
    `
        : ''
    }

    ${(() => {
      switch ($variant) {
        case 'p':
          return `
            color: inherit;
            line-height: 1.5;
          `;
        case 'lead':
          return `
            font-size: 1.25rem;
            font-weight: 300;
            line-height: 1.5;
          `;
        default:
          return `
            color: inherit;
            font-family: 'Cabin', sans-serif;
            font-size: ${(() => {
              switch ($variant) {
                case 'h6':
                  return '1rem';
                case 'h5':
                  return '1.25rem';
                case 'h4':
                  return '1.5rem';
                case 'h3':
                  return '1.75rem';
                case 'h2':
                  return '2rem';
                default:
                  return '2.5rem';
              }
            })()};
            font-weight: 600;
            line-height: 1.2;
        `;
      }
    })()}
  `}
`;

const margins = {
  lg: 1,
  md: 0.5,
  none: 0,
  sm: 0.25,
};

interface ILineProps {
  $margin: keyof typeof margins;
}

const Line = styled.hr<ILineProps>`
  ${({ $margin }) => `
    border: 0;
    border-top: 2.5px solid #007fa8;
    margin: ${margins[$margin] ? `${margins[$margin]}rem 0` : '0'};
    padding: 0;

  `}
`;

export interface ITypographyProps {
  align?: 'left' | 'right' | 'center';
  breakWord?: boolean;
  children: ReactNode;
  className?: string;
  component: TypographyVariants;
  id?: string;
  margin?: keyof typeof margins;
  rule?: boolean;
  truncate?: boolean;
  variant: TypographyVariants | 'lead';
}

const Typography: FC<ITypographyProps> = ({
  align = 'left',
  breakWord = false,
  children,
  className = '',
  component,
  id = undefined,
  margin = 'md',
  rule = false,
  truncate = false,
  variant,
}) => (
  <>
    <BaseTypography
      id={id}
      as={component}
      className={className}
      $align={align}
      $breakWord={breakWord}
      $margin={margin}
      $truncate={truncate}
      $variant={variant}
    >
      {children}
    </BaseTypography>

    {rule && <Line $margin={margin} />}
  </>
);

export default memo(Typography);
