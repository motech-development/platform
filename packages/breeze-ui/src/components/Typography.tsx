import { FC, ReactNode } from 'react';
import {
  classNames,
  spacingClass,
  textAlignmentClass,
} from '../utils/className';
import TAlignment from '../utils/alignment';
import TSpacing from '../utils/spacing';
import Element from './Element';

export enum TypographyComponent {
  H1 = 'h1',
  H2 = 'h2',
  H3 = 'h3',
  H4 = 'h4',
  H5 = 'h5',
  H6 = 'h6',
  P = 'p',
}

export enum TypographyVariant {
  H1 = 'h1',
  H2 = 'h2',
  H3 = 'h3',
  H4 = 'h4',
  H5 = 'h5',
  H6 = 'h6',
  P = 'p',
  Lead = 'lead',
}

export type TTypographyComponent = `${TypographyComponent}`;

export type TTypographyVariant = `${TypographyVariant}`;

export interface ITypographyProps {
  align?: TAlignment;
  breakWord?: boolean;
  children: ReactNode;
  className?: string;
  component: TTypographyComponent;
  id?: string;
  margin?: TSpacing;
  rule?: boolean;
  truncate?: boolean;
  variant: TTypographyVariant;
}

const variantStyles = (variant: TTypographyVariant) => {
  switch (variant) {
    case 'h1':
      return 'font-display text-5xl font-semibold';
    case 'h2':
      return 'font-display text-4xl font-semibold';
    case 'h3':
      return 'font-display text-3xl font-semibold';
    case 'h4':
      return 'font-display text-2xl font-semibold';
    case 'h5':
      return 'font-display text-xl font-semibold';
    case 'h6':
      return 'font-display text-lg font-semibold';
    case 'lead':
      return 'font-light text-xl';
    default:
      return 'font-sans';
  }
};

const ruleAlignmentClass = (alignment: TAlignment) => {
  switch (alignment) {
    case 'center':
      return 'm-auto';
    case 'right':
      return 'ml-auto';
    default:
      return 'mr-auto';
  }
};

const ruleSpacingClass = (spacing: TSpacing) => {
  switch (spacing) {
    case 'lg':
      return 'mb-2.5';
    case 'md':
      return 'mb-1.5';
    case 'sm':
      return 'mb-0.5';
    default:
      return '';
  }
};

const Typography: FC<ITypographyProps> = ({
  align = 'left',
  breakWord = false,
  children,
  className = '',
  component,
  id,
  margin = 'md',
  rule = false,
  truncate = false,
  variant,
}) => (
  <div>
    {/* @tailwind: mb-0 mb-2 mb-4 mb-6 */}
    <Element
      id={id}
      className={classNames(
        className,
        breakWord ? 'break-words' : '',
        truncate ? 'truncate' : '',
        rule ? ruleSpacingClass(margin) : spacingClass(margin, 'mb-{spacing}'),
        textAlignmentClass(align),
        variantStyles(variant),
      )}
      as={component}
    >
      {children}
    </Element>

    {rule && (
      <hr
        className={classNames(
          'w-12 border-t-4 border-red-600 shadow-md',
          ruleAlignmentClass(align),
          ruleSpacingClass(margin),
        )}
      />
    )}
  </div>
);

export default Typography;
