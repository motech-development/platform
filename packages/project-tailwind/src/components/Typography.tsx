import { ElementType } from 'react';
import { Box, PolymorphicComponentProps } from 'react-polymorphic-box';
import {
  Alignment,
  Sizing,
  TAlignment,
  Themes,
  TSizing,
  TTheme,
  useTailwind,
} from '../utilities/tailwind';

/** Variant options */
enum Variants {
  /** Body */
  BODY = 'body',
  /** Heading 1 */
  HEADING_1 = 'h1',
  /** Heading 2 */
  HEADING_2 = 'h2',
  /** Heading 3 */
  HEADING_3 = 'h3',
  /** Heading 4 */
  HEADING_4 = 'h4',
  /** Heading 5 */
  HEADING_5 = 'h5',
  /** Heading 6 */
  HEADING_6 = 'h6',
  /** Lead */
  LEAD = 'lead',
}

/** Variant type */
type TVariant = `${Variants}`;

/** Default component element */
const DEFAULT_ELEMENT = 'p';

/** Typography component props */
export type TTypographyProps<E extends ElementType> = PolymorphicComponentProps<
  E,
  {
    /** Text alignment */
    align?: TAlignment;

    /** Apply line break for overflowing text */
    breakWord?: boolean;

    /** Apply margin to text */
    margin?: TSizing;

    /** Add horizontal rule */
    rule?: boolean;

    /** Component theme */
    theme?: TTheme;

    /** Truncate text string */
    truncate?: boolean;

    /** Text variant */
    variant?: TVariant;
  }
>;

/**
 * Creates font styles for text variants
 *
 * @param variant - Text variant type
 *
 * @returns Class names
 */
function createFontStyles(variant: TVariant) {
  switch (variant) {
    case Variants.HEADING_1:
      return 'font-display text-5xl font-semibold';
    case Variants.HEADING_2:
      return 'font-display text-4xl font-semibold';
    case Variants.HEADING_3:
      return 'font-display text-3xl font-semibold';
    case Variants.HEADING_4:
      return 'font-display text-2xl font-semibold';
    case Variants.HEADING_5:
      return 'font-display text-xl font-semibold';
    case Variants.HEADING_6:
      return 'font-display text-lg font-semibold';
    case Variants.LEAD:
      return 'font-light text-xl';
    case Variants.BODY:
    default:
      return 'font-sans';
  }
}

/**
 * Creates spacing classes for horizontal rule
 *
 * @param spacing - Spacing size
 *
 * @returns Class names
 */
function createRuleSpacingStyles(rule: boolean, spacing: TSizing) {
  switch (spacing) {
    case Sizing.LG:
      return rule ? 'mb-2.5' : 'mb-6';
    case Sizing.SM:
      return rule ? 'mb-0.5' : 'mb-2';
    case Sizing.MD:
    default:
      return rule ? 'mb-1.5' : 'mb-4';
  }
}

/**
 * Creates alignment classes for horizontal rule
 *
 * @param alignment - Horizontal rule alignment
 *
 * @returns Class names
 */
function createRuleAlignmentStyles(alignment: TAlignment) {
  switch (alignment) {
    case Alignment.CENTRE:
      return 'm-auto';
    case Alignment.RIGHT:
      return 'ml-auto';
    case Alignment.LEFT:
    default:
      return 'mr-auto';
  }
}

/**
 * Standardised typography
 *
 * @param props - Typography component props
 *
 * @returns Typography component
 */
export function Typography<E extends ElementType = typeof DEFAULT_ELEMENT>({
  align = Alignment.LEFT,
  breakWord = false,
  className,
  margin = Sizing.MD,
  rule = false,
  theme = Themes.DANGER,
  truncate = false,
  variant = Variants.BODY,
  ...rest
}: TTypographyProps<E>) {
  const { createStyles, createTextAlignmentStyles } = useTailwind(
    theme,
    margin,
  );

  const ruleStyles = createStyles({
    classNames: [
      'w-12 border-t-4 shadow-md',
      createRuleSpacingStyles(rule, margin),
      createRuleAlignmentStyles(align),
    ],
    theme: {
      danger: ['border-red-600'],
      primary: ['border-blue-600'],
      secondary: ['border-gray-600'],
      success: ['border-green-600'],
      warning: ['border-yellow-600'],
    },
  });

  const typographyStyles = createStyles({
    classNames: [
      {
        'bread-words': breakWord,
        truncate,
      },
      createFontStyles(variant),
      createTextAlignmentStyles(align),
      createRuleSpacingStyles(rule, margin),
      className,
    ],
  });

  return (
    <div>
      {/* eslint-disable-next-line react/jsx-props-no-spreading */}
      <Box className={typographyStyles} as={DEFAULT_ELEMENT} {...rest} />

      {rule && <hr className={ruleStyles} />}
    </div>
  );
}
