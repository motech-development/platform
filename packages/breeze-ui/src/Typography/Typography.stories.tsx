import { withA11y } from '@storybook/addon-a11y';
import { boolean, select, text, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React from 'react';
import BaseStyles from '../BaseStyles/BaseStyles';
import Card from '../Card/Card';
import Typography from './Typography';

type Components = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'p';

const stories = storiesOf('Typography', module);
const components = {
  H1: 'h1',
  H2: 'h2',
  H3: 'h3',
  H4: 'h4',
  H5: 'h5',
  H6: 'h6',

  Paragraph: 'p',
};
const variants = {
  ...components,
  Lead: 'lead',
};
const alignment = {
  Centre: 'center',
  Left: 'left',
  Right: 'right',
};
const margin = {
  Large: 'lg',
  Medium: 'md',
  None: 'none',
  Small: 'sm',
};

stories.addDecorator(withA11y);
stories.addDecorator(withKnobs);

stories.add('Basic typography', () => (
  <>
    <BaseStyles />

    <Card>
      <Typography
        align={
          select('Alignment', alignment, 'left') as 'left' | 'right' | 'center'
        }
        variant={select('Variant', variants, 'h1') as Components | 'lead'}
        component={select('Component', components, 'h1') as Components}
        margin={select('Margin', margin, 'md') as 'lg' | 'md' | 'sm' | 'none'}
        rule={boolean('Horizontal rule', false)}
      >
        {text('Text', 'Hello world')}
      </Typography>
    </Card>
  </>
));
