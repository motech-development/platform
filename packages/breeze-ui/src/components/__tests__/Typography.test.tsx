import { render } from '@testing-library/react';
import Typography from '../Typography';

describe('Typography', () => {
  describe('component', () => {
    it('should render an h1 element', () => {
      const { container } = render(
        <Typography component="h1" variant="h1">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render an h2 element', () => {
      const { container } = render(
        <Typography component="h2" variant="h2">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render an h3 element', () => {
      const { container } = render(
        <Typography component="h3" variant="h3">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render an h4 element', () => {
      const { container } = render(
        <Typography component="h4" variant="h4">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render an h5 element', () => {
      const { container } = render(
        <Typography component="h5" variant="h5">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render an h6 element', () => {
      const { container } = render(
        <Typography component="h6" variant="h6">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render an p element', () => {
      const { container } = render(
        <Typography component="p" variant="p">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('variant', () => {
    it('should have the correct h1 styles', () => {
      const { container } = render(
        <Typography component="h1" variant="h1">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should have the correct h2 styles', () => {
      const { container } = render(
        <Typography component="h2" variant="h2">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should have the correct h3 styles', () => {
      const { container } = render(
        <Typography component="h3" variant="h3">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should have the correct h4 styles', () => {
      const { container } = render(
        <Typography component="h4" variant="h4">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should have the correct h5 styles', () => {
      const { container } = render(
        <Typography component="h5" variant="h5">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should have the correct h6 styles', () => {
      const { container } = render(
        <Typography component="h6" variant="h6">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should have the correct p styles', () => {
      const { container } = render(
        <Typography component="p" variant="p">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should have the correct lead styles', () => {
      const { container } = render(
        <Typography component="p" variant="lead">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('alignment', () => {
    it('should align to the left by default', () => {
      const { container } = render(
        <Typography component="h1" variant="h1">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should align to the left', () => {
      const { container } = render(
        <Typography align="left" component="h1" variant="h1">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should align to the right', () => {
      const { container } = render(
        <Typography align="right" component="h1" variant="h1">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should align to the centre', () => {
      const { container } = render(
        <Typography align="center" component="h1" variant="h1">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('margin', () => {
    it('should have the correct margin by default', () => {
      const { container } = render(
        <Typography component="h1" variant="h1">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should have the correct margin when set to none', () => {
      const { container } = render(
        <Typography component="h1" variant="h1" margin="none">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should have the correct margin when set to sm', () => {
      const { container } = render(
        <Typography component="h1" variant="h1" margin="sm">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should have the correct margin when set to md', () => {
      const { container } = render(
        <Typography component="h1" variant="h1" margin="md">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should have the correct margin when set to lg', () => {
      const { container } = render(
        <Typography component="h1" variant="h1" margin="lg">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('rule', () => {
    it('should display a horizontal rule', () => {
      const { container } = render(
        <Typography rule component="h1" variant="h1" margin="lg">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should have the correct margin by default', () => {
      const { container } = render(
        <Typography rule component="h1" variant="h1">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should have the correct margin when set to none', () => {
      const { container } = render(
        <Typography rule component="h1" variant="h1" margin="none">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should have the correct margin when set to sm', () => {
      const { container } = render(
        <Typography rule component="h1" variant="h1" margin="sm">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should have the correct margin when set to md', () => {
      const { container } = render(
        <Typography rule component="h1" variant="h1" margin="md">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should have the correct margin when set to lg', () => {
      const { container } = render(
        <Typography rule component="h1" variant="h1" margin="lg">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should not display a horizontal rule', () => {
      const { container } = render(
        <Typography rule={false} component="h1" variant="h1" margin="lg">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should not display a horizontal rule by default', () => {
      const { container } = render(
        <Typography component="h1" variant="h1" margin="lg">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('id', () => {
    it('should not have an id by default', () => {
      const { container } = render(
        <Typography component="h1" variant="h1" margin="lg">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should have an id if set', () => {
      const { container } = render(
        <Typography id="test" component="h1" variant="h1" margin="lg">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('break word', () => {
    it('should set the additional word break styles', () => {
      const { container } = render(
        <Typography breakWord component="h1" variant="h1">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('truncate', () => {
    it('should set the additional truncate styles', () => {
      const { container } = render(
        <Typography truncate component="h1" variant="h1">
          Hello world
        </Typography>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
