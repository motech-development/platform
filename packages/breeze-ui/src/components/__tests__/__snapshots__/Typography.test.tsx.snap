// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography alignment should align to the centre 1`] = `
<div>
  <h1
    class="mb-4 text-center font-display text-5xl font-semibold"
  >
    Hello world
  </h1>
</div>
`;

exports[`Typography alignment should align to the left 1`] = `
<div>
  <h1
    class="mb-4 text-left font-display text-5xl font-semibold"
  >
    Hello world
  </h1>
</div>
`;

exports[`Typography alignment should align to the left by default 1`] = `
<div>
  <h1
    class="mb-4 text-left font-display text-5xl font-semibold"
  >
    Hello world
  </h1>
</div>
`;

exports[`Typography alignment should align to the right 1`] = `
<div>
  <h1
    class="mb-4 text-right font-display text-5xl font-semibold"
  >
    Hello world
  </h1>
</div>
`;

exports[`Typography break word should set the additional word break styles 1`] = `
<div>
  <h1
    class="break-words mb-4 text-left font-display text-5xl font-semibold"
  >
    Hello world
  </h1>
</div>
`;

exports[`Typography component should render an h1 element 1`] = `
<div>
  <h1
    class="mb-4 text-left font-display text-5xl font-semibold"
  >
    Hello world
  </h1>
</div>
`;

exports[`Typography component should render an h2 element 1`] = `
<div>
  <h2
    class="mb-4 text-left font-display text-4xl font-semibold"
  >
    Hello world
  </h2>
</div>
`;

exports[`Typography component should render an h3 element 1`] = `
<div>
  <h3
    class="mb-4 text-left font-display text-3xl font-semibold"
  >
    Hello world
  </h3>
</div>
`;

exports[`Typography component should render an h4 element 1`] = `
<div>
  <h4
    class="mb-4 text-left font-display text-2xl font-semibold"
  >
    Hello world
  </h4>
</div>
`;

exports[`Typography component should render an h5 element 1`] = `
<div>
  <h5
    class="mb-4 text-left font-display text-xl font-semibold"
  >
    Hello world
  </h5>
</div>
`;

exports[`Typography component should render an h6 element 1`] = `
<div>
  <h6
    class="mb-4 text-left font-display text-lg font-semibold"
  >
    Hello world
  </h6>
</div>
`;

exports[`Typography component should render an p element 1`] = `
<div>
  <p
    class="mb-4 text-left font-sans"
  >
    Hello world
  </p>
</div>
`;

exports[`Typography id should have an id if set 1`] = `
<div>
  <h1
    class="mb-6 text-left font-display text-5xl font-semibold"
    id="test"
  >
    Hello world
  </h1>
</div>
`;

exports[`Typography id should not have an id by default 1`] = `
<div>
  <h1
    class="mb-6 text-left font-display text-5xl font-semibold"
  >
    Hello world
  </h1>
</div>
`;

exports[`Typography margin should have the correct margin by default 1`] = `
<div>
  <h1
    class="mb-4 text-left font-display text-5xl font-semibold"
  >
    Hello world
  </h1>
</div>
`;

exports[`Typography margin should have the correct margin when set to lg 1`] = `
<div>
  <h1
    class="mb-6 text-left font-display text-5xl font-semibold"
  >
    Hello world
  </h1>
</div>
`;

exports[`Typography margin should have the correct margin when set to md 1`] = `
<div>
  <h1
    class="mb-4 text-left font-display text-5xl font-semibold"
  >
    Hello world
  </h1>
</div>
`;

exports[`Typography margin should have the correct margin when set to none 1`] = `
<div>
  <h1
    class="mb-0 text-left font-display text-5xl font-semibold"
  >
    Hello world
  </h1>
</div>
`;

exports[`Typography margin should have the correct margin when set to sm 1`] = `
<div>
  <h1
    class="mb-2 text-left font-display text-5xl font-semibold"
  >
    Hello world
  </h1>
</div>
`;

exports[`Typography rule should display a horizontal rule 1`] = `
<div>
  <h1
    class="mb-2.5 text-left font-display text-5xl font-semibold"
  >
    Hello world
  </h1>
  <hr
    class="w-12 border-t-4 border-red-600 shadow-md mr-auto mb-2.5"
  />
</div>
`;

exports[`Typography rule should have the correct margin by default 1`] = `
<div>
  <h1
    class="mb-1.5 text-left font-display text-5xl font-semibold"
  >
    Hello world
  </h1>
  <hr
    class="w-12 border-t-4 border-red-600 shadow-md mr-auto mb-1.5"
  />
</div>
`;

exports[`Typography rule should have the correct margin when set to lg 1`] = `
<div>
  <h1
    class="mb-2.5 text-left font-display text-5xl font-semibold"
  >
    Hello world
  </h1>
  <hr
    class="w-12 border-t-4 border-red-600 shadow-md mr-auto mb-2.5"
  />
</div>
`;

exports[`Typography rule should have the correct margin when set to md 1`] = `
<div>
  <h1
    class="mb-1.5 text-left font-display text-5xl font-semibold"
  >
    Hello world
  </h1>
  <hr
    class="w-12 border-t-4 border-red-600 shadow-md mr-auto mb-1.5"
  />
</div>
`;

exports[`Typography rule should have the correct margin when set to none 1`] = `
<div>
  <h1
    class="text-left font-display text-5xl font-semibold"
  >
    Hello world
  </h1>
  <hr
    class="w-12 border-t-4 border-red-600 shadow-md mr-auto"
  />
</div>
`;

exports[`Typography rule should have the correct margin when set to sm 1`] = `
<div>
  <h1
    class="mb-0.5 text-left font-display text-5xl font-semibold"
  >
    Hello world
  </h1>
  <hr
    class="w-12 border-t-4 border-red-600 shadow-md mr-auto mb-0.5"
  />
</div>
`;

exports[`Typography rule should not display a horizontal rule 1`] = `
<div>
  <h1
    class="mb-6 text-left font-display text-5xl font-semibold"
  >
    Hello world
  </h1>
</div>
`;

exports[`Typography rule should not display a horizontal rule by default 1`] = `
<div>
  <h1
    class="mb-6 text-left font-display text-5xl font-semibold"
  >
    Hello world
  </h1>
</div>
`;

exports[`Typography truncate should set the additional truncate styles 1`] = `
<div>
  <h1
    class="truncate mb-4 text-left font-display text-5xl font-semibold"
  >
    Hello world
  </h1>
</div>
`;

exports[`Typography variant should have the correct h1 styles 1`] = `
<div>
  <h1
    class="mb-4 text-left font-display text-5xl font-semibold"
  >
    Hello world
  </h1>
</div>
`;

exports[`Typography variant should have the correct h2 styles 1`] = `
<div>
  <h2
    class="mb-4 text-left font-display text-4xl font-semibold"
  >
    Hello world
  </h2>
</div>
`;

exports[`Typography variant should have the correct h3 styles 1`] = `
<div>
  <h3
    class="mb-4 text-left font-display text-3xl font-semibold"
  >
    Hello world
  </h3>
</div>
`;

exports[`Typography variant should have the correct h4 styles 1`] = `
<div>
  <h4
    class="mb-4 text-left font-display text-2xl font-semibold"
  >
    Hello world
  </h4>
</div>
`;

exports[`Typography variant should have the correct h5 styles 1`] = `
<div>
  <h5
    class="mb-4 text-left font-display text-xl font-semibold"
  >
    Hello world
  </h5>
</div>
`;

exports[`Typography variant should have the correct h6 styles 1`] = `
<div>
  <h6
    class="mb-4 text-left font-display text-lg font-semibold"
  >
    Hello world
  </h6>
</div>
`;

exports[`Typography variant should have the correct lead styles 1`] = `
<div>
  <p
    class="mb-4 text-left font-light text-xl"
  >
    Hello world
  </p>
</div>
`;

exports[`Typography variant should have the correct p styles 1`] = `
<div>
  <p
    class="mb-4 text-left font-sans"
  >
    Hello world
  </p>
</div>
`;
